code/include/playfab_c/PlayFabClientApi_c.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include "httpClient/pal.h"
+#include "httpClient/async.h"
+#include "PlayFabClientDataModels_c.h"
+
+/// <summary>
+/// Initialize PlayFab global state
+/// </summary>
+HRESULT PlayFabInitialize(
+    _In_ const char* titleId
+) noexcept;
+
+/// <summary>
+/// Cleanup PlayFab global state. Shouldn't be called while async APIs are still running.
+/// </summary>
+HRESULT PlayFabCleanupAsync(
+    _In_ XAsyncBlock* async
+) noexcept;
+
+/// <summary>
+/// Get the result handle for an asynchronous API
+/// </summary>
+HRESULT PlayFabGetResultHandle(
+    _In_ XAsyncBlock* async,
+    _Out_ PlayFabResultHandle* resultHandle
+) noexcept;
+
+/// <summary>
+/// Duplicate result.
+/// </summary>
+HRESULT PlayFabResultDuplicateHandle(
+    _In_ PlayFabResultHandle resultHandle,
+    _Out_ PlayFabResultHandle* duplicatedHandle
+) noexcept;
+
+/// <summary>
+/// Close result. Invalidates all pointers associated with the result handle.
+/// </summary>
+void PlayFabResultCloseHandle(
+    _In_ PlayFabResultHandle resultHandle
+) noexcept;
+

```

Comment:
these are generic so PlayFabResultHandle is reused for each PF API.  I'm not sure GameCore review will like this as its not strongly typed result handle.  

aka inside async callback of PlayFabLoginWithCustomIDAsync(), you call PlayFabGetResultHandle() to get PlayFabResultHandle.  And then its not clear to caller that they shouldn't call 

HRESULT PlayFabGetPlayerProfileResultGetProfile(
    _In_ PlayFabResultHandle resultHandle,
    _Out_ PlayFabPlayerProfileModel** profile
) noexcept;

same goes for other combos of apis.  aka they need to read docs of which combos of gets are valid for each API call

Reply:
Ya it might be worth defining a unique result handle type (and GetResultHandle API) for each *Aysnc API if it makes this more obvious.

---
In reply to: [590777452](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590777452) [](ancestors = 590777452)

Reply:
Changed design slightly and hopefully this is a non issue now.

---
In reply to: [590803429](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590803429) [](ancestors = 590803429,590777452)